specs/language/introduction.tex

```diff
@@ -273,21 +273,52 @@
 
 \Sec{\acrshort{hlsl} Memory Models}{Intro.Memory}
 
-\p Memory accesses for \gls{sm} 5.0 and earlier operate on 128-bit slots aligned
-on 128-bit boundaries. This optimized for the common case in early shaders where
-data being processed on the GPU was usually 4-element vectors of 32-bit data
-types.
-
-\p On modern hardware memory access restrictions are loosened, and reads of
-32-bit multiples are supported starting with \gls{sm} 5.1 and reads of 16-bit
-multiples are supported with \gls{sm} 6.0. \gls{sm} features are fully
-documented in the \gls{dx} Specifications, and this document will not attempt to
-elaborate further.
+\p The fundamental storage unit in HLSL is a \textit{byte}, which is comprised
+of 8 \textit{bits}. Each \textit{bit} stores a single value 0 or 1. Each byte
+has a unique \textit{memory location}, alternatively called an \textit{address}.
+
+\p An object that is not of an intangible type will have an associated set of
+memory locations containing one or more location. For a bit-field the location
+will represent a maximal sequence of adjacent bit-fields all having nonzero
+width.
+
+\p Each read or write to a memory location is called a \textit{memory access}.
+Operations that perform memory accesses are called \textit{memory operation}. A
+memory operation may operate on one or more memory locations. A memory operation
+must not alter memory at a location not contained in the memory location set it
+is operating on\footnote{Two subtle notes here: (1) A bit-field's memory location
+includes adjacent bit-fields, so reads and writes to bit-fields are expected to
+read and write adjacent memory if they're within the same set of locations, (2)
+padding bits inside a structure are included in the memory location of the
+structure. Reads and writes of uninitialized memory may be undefined, but a
+write is allowed to stomp over padding.}.
```

Comment:
>  Reads and writes of uninitialized memory may be undefined, but a write is allowed to stomp over padding.

Does "undefined here" mean "undefined behavior"?  I'm also wondering about the "may" - is this a decision that hasn't been made yet? Or are there some cases where it is defined?

I'm also intrigued why writing to uninitialized memory could be undefined behavior, if that's what this means. Surely that's how you initialize the memory!

Reply:
Reading and writing _of_ uninitialized memory (not to) is undefined. Basically if you allocate an `int` but don't set it, reading that value is undefined, writing the undefined value you read to memory is undefined.

I probably should say it `is` undefined, rather than `may be`.